src/tls.rs

```diff
@@ -320,6 +327,11 @@ impl AsyncRead for TlsStream {
         match pin.state {
             State::Handshaking(ref mut accept) => match ready!(Pin::new(accept).poll(cx)) {
                 Ok(mut stream) => {
+                    let (_, conn) = stream.get_ref();
+                    *pin.peer_certs.write().unwrap() = conn
+                        .peer_certificates()
+                        .map(|certs| peer_certs_into_owned(&certs.to_vec()));
```

Comment:
```suggestion
                        .map(|certs| certs.to_vec());
```
this passes tests.

Reply:
This doesn't pass for me. Maybe the tls feature flag again?